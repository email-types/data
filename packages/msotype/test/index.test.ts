import path from 'path';
import ts from 'typescript';

const typescriptDiagnostic = (
  fileName: string[],
  options: ts.CompilerOptions,
): string[] => {
  const program = ts.createProgram(fileName, options);
  const diagnostics = ts.getPreEmitDiagnostics(program);

  return diagnostics.map(({ file, start = 0, messageText }) => {
    const diagFile = file && file.getLineAndCharacterOfPosition(start);

    if (diagFile) {
      return `${diagFile.line}:${
        diagFile.character
      } - ${ts.flattenDiagnosticMessageText(messageText, '\n')}`;
    }

    return `${ts.flattenDiagnosticMessageText(messageText, '\n')}`;
  });
};

it('throws typescript errors', () => {
  const result = typescriptDiagnostic(
    [path.resolve(__dirname, 'fixtures/hasTsErrors.ts')],
    { noEmit: true, strict: true },
  );

  expect(result).toHaveLength(2);
  expect(result[0]).toMatchSnapshot();
  expect(result[1]).toMatchSnapshot();
});

it('throws no typescript errors', () => {
  const result = typescriptDiagnostic(
    [path.resolve(__dirname, 'fixtures/noTsErrors.ts')],
    { noEmit: true, strict: true },
  );

  expect(result).toHaveLength(0);
});
