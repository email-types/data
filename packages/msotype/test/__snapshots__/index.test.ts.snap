// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws typescript errors 1`] = `
"4:2 - Type '{ unknownProperty: number; }' is not assignable to type 'Properties<0 | GlobalsString>'.
  Object literal may only specify known properties, and 'unknownProperty' does not exist in type 'Properties<0 | GlobalsString>'."
`;

exports[`throws typescript errors 2`] = `"8:2 - Type 'number' is not assignable to type '\\"at-least\\" | \\"exactly\\" | undefined'."`;
